pennylane_lightning/lightning_tensor/_measurements.py

```diff
@@ -207,9 +215,162 @@ def measure_tensor_network(self, circuit: QuantumScript) -> Result:
         """
 
         if circuit.shots:
-            raise NotImplementedError("Shots are not supported for tensor network simulations.")
+            # finite-shot case
+            results = self.measure_with_samples(
+                circuit.measurements,
+                shots=circuit.shots,
+            )
+
+            if len(circuit.measurements) == 1:
+                if circuit.shots.has_partitioned_shots:
+                    return tuple(res[0] for res in results)
+
+                return results[0]
+
+            return results
         # analytic case
         if len(circuit.measurements) == 1:
             return self.measurement(circuit.measurements[0])
 
         return tuple(self.measurement(mp) for mp in circuit.measurements)
+
+    # pylint:disable = too-many-arguments
+    def measure_with_samples(
+        self,
+        measurements: List[Union[SampleMeasurement, ClassicalShadowMP, ShadowExpvalMP]],
+        shots: Shots,
+    ) -> List[TensorLike]:
+        """
+        Returns the samples of the measurement process performed on the given state.
+        This function assumes that the user-defined wire labels in the measurement process
+        have already been mapped to integer wires used in the device.
+
+        Args:
+            measurements (List[Union[SampleMeasurement, ClassicalShadowMP, ShadowExpvalMP]]):
+                The sample measurements to perform
+            shots (Shots): The number of samples to take
+
+        Returns:
+            List[TensorLike[Any]]: Sample measurement results
+        """
+        mps = measurements
+        groups, indices = _group_measurements(mps)
+
+        all_res = []
+        for group in groups:
+            if isinstance(group[0], (ExpectationMP, VarianceMP)) and isinstance(
+                group[0].obs, SparseHamiltonian
+            ):
+                raise TypeError(
+                    "ExpectationMP/VarianceMP(SparseHamiltonian) cannot be computed with samples."
+                )
+            if isinstance(group[0], VarianceMP) and isinstance(group[0].obs, (Hamiltonian, Sum)):
+                raise TypeError("VarianceMP(Hamiltonian/Sum) cannot be computed with samples.")
+            if isinstance(group[0], (ClassicalShadowMP, ShadowExpvalMP)):
+                raise TypeError(
+                    "ExpectationMP(ClassicalShadowMP, ShadowExpvalMP) cannot be computed with samples."
+                )
```

Comment:
Can these be combined, using `type(group[0]).__name__` in an f-string for the error message?